cmd/sos_show.go

```diff
@@ -0,0 +1,187 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/exoscale/cli/table"
+	"github.com/spf13/cobra"
+)
+
+type sosACLShowOutput struct {
+	Access string `json:"access"`
+	Value  string `json:"value"`
+}
+
+type sosMetadataShowOutput struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type sosHeadersShowOutput struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type sosShowOutput struct {
+	Object   string                  `json:"object"`
+	URL      string                  `json:"url"`
+	ACL      []sosACLShowOutput      `json:"acl"`
+	Metadata []sosMetadataShowOutput `json:"metadata"`
+	Headers  []sosHeadersShowOutput  `json:"headers"`
+}
+
+func (o *sosShowOutput) toJSON() { outputJSON(o) }
+
+func (o *sosShowOutput) toText() { outputText(o) }
+
+func (o *sosShowOutput) toTable() {
+	t := table.NewTable(os.Stdout)
+	t.SetHeader([]string{o.Object})
+
+	if o.ACL != nil {
+		buf := bytes.NewBuffer(nil)
+		at := table.NewEmbeddedTable(buf)
+		at.SetHeader([]string{"access", "value"})
```

Comment:
I'm not sure this format is suitable for canned ACLs: when an object is applied a canned ACL, I suggest simplifying the output to the canned ACL name only, and switch to "detailed" table-based output only if the object has complex manually set ACLs.